src/Tooltip/Tooltip.tsx

```diff
@@ -53,68 +112,162 @@ export class Tooltip extends React.Component<TooltipProps, TooltipState> {
-
-                    {!this.props.messageGroup &&
-                        <div className="message-container" style={{ width: `${this.props.width || 120}px` }}>
-                            {this.props.title && <div className="title">{this.props.title}</div>}
-                            <div className="message">{this.props.message || "Tooltip is empty. Please pass a message."}</div>
-                        </div>
-                    }
-
-                    {this.props.messageGroup &&
-                        <div className="message-container" style={{ width: `${this.props.width || 120}px` }}>
-                            {this.props.messageGroup.map((item, index) =>
-                                <div key={index} className="message-list-item">
-                                    {item.title && <div className="title">{item.title}</div>}
-                                    {item.message && <div className="message">{item.message}</div>}
-                                </div>
-                            )}
-                        </div>
-                    }
-
-                    {(this.isPositioned("top") || this.isPositioned("right") || this.isPositioned("left")) && <div className="triangle" />}
+                    {this.props.children || <div className="icon">{this.props.customSvg ? this.props.customSvg : InfoCircleIcon}</div>}
                 </div>
             </div>
         );
     }
+
+    /**
+     * construct tooltip content container and append to root tooltip container
+     * @param tooltipId unique tooltip id
+     */
+    private constructTooltipContentContainer = (tooltipId: string): void => {
+        const tooltipRootArr: HTMLCollectionOf<Element> = document.body.getElementsByClassName(this.tooltipRootClassName);
+        let tooltipRootRef: Element = tooltipRootArr && tooltipRootArr[0];
+        if (tooltipRootArr.length === 0) {
+            tooltipRootRef = document.createElement("div");
+            tooltipRootRef.className = this.tooltipRootClassName;
+            document.body.appendChild(tooltipRootRef);
+        }
+        const newTooltip: HTMLDivElement = document.createElement("div");
+        newTooltip.className = `tooltip-content ${this.props.theme || "default"} ${this.props.position}`;
+        newTooltip.id = tooltipId;
+        newTooltip.setAttribute("role", "tooltip");
+        this.setTooltipContent(newTooltip);
+        tooltipRootRef.appendChild(newTooltip);
+        this.setState({
+            tooltipContainer: newTooltip,
+            tooltipPositionChecker: new TooltipPositionChecker(this.containerRef.current, this.props.disableAutoPosition)
+        }, () => {
+            this.state.tooltipPositionChecker.addTooltipContainer(this.state.tooltipContainer);
+        });
+    }
+
+    /**
+     * construct unique tooltip id
+     */
+    private constructId = (): string => {
+        const randId: string = randomId("tooltip-");
```

Comment:
i believe `ramdomId` here comes from another dependency right ? unless its mention as a peer-dependency in this project, it may raise an issue;

Reply:
randomId is in __utils folder